### PR TITLE
Upgrade zookeeper

### LINKED ISSUE #300: 🔐 CVE-2024-23944: org.apache.zookeeper:zookeeper:jar:3.9.1:compile
## Summary
Information disclosure in persistent watchers handling in Apache ZooKeeper due to missing ACL check. It allows an attacker to monitor child znodes by attaching a persistent watcher (addWatch command) to a parent which the attacker has already access to. ZooKeeper server doesn't do ACL check when the persistent watcher is triggered and as a consequence, the full path of znodes that a watch event gets triggered upon is exposed to the owner of the watcher. It's important to note that only the path is exposed by this vulnerability, not the data of znode, but since znode path can contain sensitive information like user name or login ID, this issue is potentially critical.

Users are recommended to upgrade to version 3.9.2, 3.8.4 which fixes the issue.

CVE: CVE-2024-23944
CWE: CWE-200

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-23944?component-type=maven&component-name=org.apache.zookeeper%2Fzookeeper&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-23944
- https://github.com/advisories/GHSA-r978-9m6m-6gm6

### PR DESCRIPTION
**Pull request by Shmuma**
Closes #300

### CODE REVIEW

**Review by Nicoretti — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -178,7 +178,7 @@
             <!-- override version 3.6.3 to fix vulnerability CVE-2023-42503 -->
```
Comment (kaklakariada) [inline pom.xml:178]:
```suggestion
            <!-- override version 3.6.3 to fix vulnerability CVE-2023-42503 and CVE-2024-23944 -->
```

File: doc/changes/changes_2.7.11.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Cloud Storage Extension 2.7.11, released 2024-03-22
+
+Code name: upgrade zookeeper to fix CVE-2024-23944
+
+## Summary
+Zookeeper dependency was upgraded to address CVE-2024-23944
+
+## Features
+
+* #300: CVE-2024-23944: org.apache.zookeeper:zookeeper:jar:3.9.1:compile
```
Comment (kaklakariada) [inline doc/changes/changes_2.7.11.md:10]:
```suggestion
* #300: Fixed CVE-2024-23944 in `org.apache.zookeeper:zookeeper:jar:3.9.1:compile`
```

We use past tense here

File: doc/changes/changes_2.7.11.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Cloud Storage Extension 2.7.11, released 2024-03-22
+
+Code name: upgrade zookeeper to fix CVE-2024-23944
+
+## Summary
+Zookeeper dependency was upgraded to address CVE-2024-23944
+
+## Features
```
Comment (kaklakariada) [inline doc/changes/changes_2.7.11.md:8]:
```suggestion
## Summary

Zookeeper dependency was upgraded to address CVE-2024-23944.

## Security
```

File: doc/changes/changes_2.7.11.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Cloud Storage Extension 2.7.11, released 2024-03-22
+
+Code name: upgrade zookeeper to fix CVE-2024-23944
```
Comment (kaklakariada) [inline doc/changes/changes_2.7.11.md:3]:
```suggestion
Code name: Upgrade zookeeper to fix CVE-2024-23944
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**